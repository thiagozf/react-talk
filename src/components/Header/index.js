import React from 'react';
import styled from 'styled-components';
import { colors } from 'components/constants';

import logo from './logo.png';

const HeaderWrapper = styled.header`
  background-color: ${colors.lightGrey};
  color: ${colors.white};
  height: 150px;
  padding: 20px;
  text-align: center;
`;

const HeaderImg = styled.img`
`;

const Header = () => (
  <HeaderWrapper>
    <HeaderImg src={logo} alt="logo" />
  </HeaderWrapper>
);

export default Header;
