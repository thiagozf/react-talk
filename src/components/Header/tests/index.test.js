import React from 'react';
import { shallow } from 'enzyme';

import Header from '../';

describe('Header tests', () => {
  it('should render', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toBeDefined();
  });

  it('should match UI defs', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
