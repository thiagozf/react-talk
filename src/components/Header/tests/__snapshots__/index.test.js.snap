// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests should match UI defs 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <styled.img
        alt="logo"
        src="logo.png"
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "alt": "logo",
        "src": "logo.png",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.img
          alt="logo"
          src="logo.png"
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "alt": "logo",
          "src": "logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
